vector/src/test/java/im/vector/app/features/home/room/detail/timeline/factory/PollItemViewStateFactoryTest.kt

```diff
@@ -0,0 +1,221 @@
+import im.vector.app.features.home.room.detail.timeline.item.PollResponseData
+import im.vector.app.features.home.room.detail.timeline.item.PollVoteSummaryData
+import im.vector.app.features.home.room.detail.timeline.item.ReactionsSummaryData
+import im.vector.app.features.home.room.detail.timeline.style.TimelineMessageLayout
+import im.vector.app.features.poll.PollViewState
+import im.vector.app.test.fakes.FakeStringProvider
+import org.amshove.kluent.shouldBeEqualTo
+import org.junit.Before
+import org.junit.Test
+import org.matrix.android.sdk.api.session.room.model.message.MessagePollContent
+import org.matrix.android.sdk.api.session.room.model.message.PollAnswer
+import org.matrix.android.sdk.api.session.room.model.message.PollCreationInfo
+import org.matrix.android.sdk.api.session.room.model.message.PollQuestion
+import org.matrix.android.sdk.api.session.room.model.message.PollType
+import org.matrix.android.sdk.api.session.room.send.SendState
+
+private val A_MESSAGE_INFORMATION_DATA = MessageInformationData(
+        eventId = "eventId",
+        senderId = "senderId",
+        ageLocalTS = 0,
+        avatarUrl = "",
+        sendState = SendState.SENT,
+        messageLayout = TimelineMessageLayout.Default(showAvatar = true, showDisplayName = true, showTimestamp = true),
+        reactionsSummary = ReactionsSummaryData(),
+        sentByMe = true,
+)
+
+private val A_POLL_RESPONSE_DATA = PollResponseData(
+        myVote = null,
+        votes = emptyMap(),
+)
+
+private val A_POLL_OPTION_IDS = listOf("5ef5f7b0-c9a1-49cf-a0b3-374729a43e76", "ec1a4db0-46d8-4d7a-9bb6-d80724715938", "3677ca8e-061b-40ab-bffe-b22e4e88fcad")
+
+private val A_POLL_CONTENT = MessagePollContent(
+        unstablePollCreationInfo = PollCreationInfo(
+                question = PollQuestion(
+                        unstableQuestion = "What is your favourite coffee?"
+                ),
+                kind = PollType.UNDISCLOSED_UNSTABLE,
+                maxSelections = 1,
+                answers = listOf(
+                        PollAnswer(
+                                id = A_POLL_OPTION_IDS[0],
+                                unstableAnswer = "Double Espresso"
+                        ),
+                        PollAnswer(
+                                id = A_POLL_OPTION_IDS[1],
+                                unstableAnswer = "Macchiato"
+                        ),
+                        PollAnswer(
+                                id = A_POLL_OPTION_IDS[2],
+                                unstableAnswer = "Iced Coffee"
+                        ),
+                )
+        )
+)
+
+class PollItemViewStateFactoryTest {
+
+    private lateinit var pollItemViewStateFactory: PollItemViewStateFactory
+
+    private val stringProvider = FakeStringProvider()
+
+    @Before
+    fun setup() {
+        // We are not going to test any UI related code
+        pollItemViewStateFactory = PollItemViewStateFactory(
```

Comment:
we can inline the creation as the junit test runner will create a new instance of the test class for each test

```
private val stringProvider = FakeStringProvider()
private val pollItemViewStateFactory = PollItemViewStateFactory(stringProvider.instance)
```